trajectory_tracker/test/CMakeLists.txt

```
@@ -1,5 +1,8 @@
 include_directories(include)
 
+# This flag is needed to compile dynamic_reconfigure::Client on Melodic
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-reorder")
```

Is it possible to do like
```suggestion
if($ENV{ROS_DISTRO} EQUAL "melodic")
  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-reorder")
endif()
```
? (not tested)

If it is a known problem of dynamic_reconfigure, is there any related upstream issue?

---

Done

It seems that it was reported but not fixed for a long time.
https://github.com/ros/dynamic_reconfigure/pull/118